lib/matplotlib/testing/conftest.py

```diff
@@ -43,7 +43,13 @@ def mpl_test_settings(request):
             # This import must come after setup() so it doesn't load the
             # default backend prematurely.
             import matplotlib.pyplot as plt
-            plt.switch_backend(backend)
+            try:
+                plt.switch_backend(backend)
+            except ImportError as exc:
+                # Should only occur for the cairo backend tests, if neither
+                # pycairo nor cairocffi are installed.
+                pytest.skip("Failed to switch to backend {} ({})."
```

Comment:
Does `exc` contain some text related to cairo? If so, I would like the skip to be conditional on that. Otherwise I'm a bit concerned that this may hide other backend problems.

Reply:
You can make that only happen if `"cairo" in backend`... feel free to push that.

Reply:
Done... Hm, technically I now cannot review anymore. Anyway, I'll approve but leave open for someone else to merge.